src/select/Option.js

```diff
@@ -56,10 +60,22 @@ class Option extends Component {
 
     const className = `crane-select-option${isFocused ? ' focused' : ''} ${allowSelectAll && option[valueKey] === allOption[valueKey] ? ' crane-select-group-header' : ''}`
 
+    let selected = false
+
+    if (value !== null) {
+      if (typeof(value) === 'string') {
```

Comment:
What if it's a number? 🤔 Would it be better to reverse this logic to cover more cases?
`
if (value) {
  if (value.length > 0 && value.some(val => isValueEqual(option, val, valueKey))) {
     selected = true
  } else if (value === option[valueKey]) {
    selected = true
  }
}
`

Reply:
Thanks, updated to include a few more scenarios.